### ISSUE TITLE
Panic on RouterContext

### ISSUE THREAD

**Issue by denyskoch** [go-chi/chi#240]
Hi, just updated chi sources and getting panic:

```
2017/08/03 23:43:55 [MacBook-Pro.local/zkvebarPeQ-000003] "GET http://localhost:8001/facebook/redirect HTTP/1.1" from [::1]:53421 - 000 0B in 5µs
2017/08/03 23:43:55 http: panic serving [::1]:53421: interface conversion: interface {} is nil, not *chi.Context
goroutine 26 [running]:
net/http.(*conn).serve.func1(0xc4201e5a40)
	/usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1721 +0xd0
panic(0x1415520, 0xc4202182c0)
	/usr/local/Cellar/go/1.8.3/libexec/src/runtime/panic.go:489 +0x2cf
github.com/go-chi/chi.RouteContext(0x166b5c0, 0xc420206270, 0x137273e)
	/Users/denyskoch/go/src/github.com/go-chi/chi/context.go:97 +0x88
github.com/pressly/chi/middleware.StripSlashes.func1(0x1b65110, 0xc420218280, 0xc42015d100)
	/Users/denyskoch/go/src/github.com/pressly/chi/middleware/strip.go:15 +0x5d
net/http.HandlerFunc.ServeHTTP(0xc4201533e0, 0x1b65110, 0xc420218280, 0xc42015d100)
	/usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
github.com/pressly/chi/middleware.RequestLogger.func1.1(0x166ad80, 0xc4201d4380, 0xc42015d000)
	/Users/denyskoch/go/src/github.com/pressly/chi/middleware/logger.go:46 +0x1b4
net/http.HandlerFunc.ServeHTTP(0xc420146f90, 0x166ad80, 0xc4201d4380, 0xc42015d000)
	/usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
github.com/pressly/chi/middleware.RealIP.func1(0x166ad80, 0xc4201d4380, 0xc42015d000)
	/Users/denyskoch/go/src/github.com/pressly/chi/middleware/realip.go:34 +0xa1
net/http.HandlerFunc.ServeHTTP(0xc420153400, 0x166ad80, 0xc4201d4380, 0xc42015d000)
	/usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
github.com/pressly/chi/middleware.RequestID.func1(0x166ad80, 0xc4201d4380, 0xc42015cf00)
	/Users/denyskoch/go/src/github.com/pressly/chi/middleware/request_id.go:68 +0x262
net/http.HandlerFunc.ServeHTTP(0xc420153420, 0x166ad80, 0xc4201d4380, 0xc42015cf00)
	/usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
github.com/pressly/chi.(*Mux).ServeHTTP(0xc42014f650, 0x166ad80, 0xc4201d4380, 0xc42015ce00)
	/Users/denyskoch/go/src/github.com/pressly/chi/mux.go:80 +0x1ea
net/http.serverHandler.ServeHTTP(0xc420190160, 0x166ad80, 0xc4201d4380, 0xc42015ce00)
	/usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:2568 +0x92
net/http.(*conn).serve(0xc4201e5a40, 0x166b500, 0xc4202180c0)
	/usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1825 +0x612
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:2668 +0x2ce

```

this panic happens when f.e. the StripSlashes middleware is used (actually when chi.RouteContext is used).

**Comment by pkieltyka** [comment]
Show sample program

**Comment by delicb** [comment]
I had same problem after upgrading to chi 3, but it turned out that I had imported middleware from `github.com/pressly/chi/middleware` (which I had in my `GOPATH`). Problem is that chi sets stuff to context and casts it when it is retrieved, but types are different because path is different, which is fine. 

Once I fixed all import paths to use `github.com/go-chi/chi`, everything worked correctly. 

@denyskoch check if same thing happened to you as well.
